packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -26,6 +43,38 @@ class TimeOfDay {
   /// The selected minute.
   final int minute;
 
+  /// Whether this time of day is before or after noon.
+  DayPeriod get period => hour < _kHoursPerPeriod ? DayPeriod.am : DayPeriod.pm;
+
+  /// Which hour of the current period (e.g., am or pm) this time is.
+  int get hourOfPeriod => hour - periodOffset;
+
+  String _addLeadingZeroIfNeeded(int value) {
+    if (value < 10)
+      return '0$value';
+    return value.toString();
+  }
+
+  /// A string representing the hour, in 24 hour time (e.g., '04' or '18').
+  String get hourLabel => _addLeadingZeroIfNeeded(hour);
+
+  /// A string representing the minute (e.g., '07').
+  String get minuteLabel => _addLeadingZeroIfNeeded(minute);
+
+  /// A string representing the hour of the current period (e.g., '4' or '6').
+  String get hourOfPeriodLabel {
+    int hourOfPeriod = this.hourOfPeriod;
```

Comment:
final, to make it clearer what's going on? It took me a few seconds to figure out why you were shadowing the getter.